### ISSUE TITLE
Stateful case insensitive completion

### ISSUE THREAD

**Issue by lenormf** [mawww/kakoune#1244]
Hi,

Often times when writing in plain english, I find myself wanting to complete upon a prefix whose full word has already been written, but in a different case. It would be very convenient if that was supported, as it currently forces me to manually modify the case of the completed word, when the current word is in lowercase.

Example:

```
Technique
techni<c-n>
```

The above would result in two `Technique` words, even though I wanted to write the second one in lower case. Maybe the completion engine could identify that I started writing the word in lowercase, and only fillin the blank (append `que`) instead of replacing the whole word?

But at the same time, sometimes we actually want this behaviour to occur, so is there a way to have the two?

Thanks.

**Comment by mawww** [comment]
I dont think the completion system should add support for that, however it should be possible to fix the case after the fact, using b\`.
